### DISCUSSION TITLE
Questions about rscp details

### DISCUSSION THREAD

**Discussion by gijoop** [anatolianroverchallenge/rscp#5] [Q&A] [answered]
Hi! Our team have a few questions related to the rscp:
1. What is the purpose of ArmDisarm message?
2. How should we interpret the integer argument in SetStage? Can we assume that there are 3 stages numbered 1 - meaning tool delivery, 2 - lava tube, 3 - rtb, or are there more stages?
3. What message type should we use to send the distance during lava tunnel stage?
4. Are you planning to extend the list of message types given in README (Acknowledge, ArmDisarm, NavigateToGPS, TaskCompleted, SetStage)? Because we can see that there are more message types in types.py

**Comment by senceryazici** [comment, marked as answer]
Hey @gijoop  !

Thanks for asking this! 

1. The `ArmDisarm` message is designed to securely arm your vehicle by sending `ArmDisarm(True)` just before the mission starts. Similarly, sending `ArmDisarm(False)` disarms your vehicle, acting as an emergency stop. This feature can be utilized during the mission to test safety mechanisms. In previous years, teams could issue software stop commands to their rovers if they detected anomalies in their algorithms. This year, the ARC judges will employ the `ArmDisarm(False)` message via the communications module to replace such e-stop behavior. (Note: This does not replace the physical emergency stop button, instead it replaces software e-stop)
2. Indeed, the autonomous mission is divided into three stages: Tool Delivery, Lava Tube Exploration, and Return to Base. I will introduce an enum to associate these stages with integer values. For your testing purposes, you can safely use 1, 2, 3 to represent these stages.
3. (and also 4.) You're correct! This repository is still being developed and will be until June 1st. The README will be updated to include all the message types. For now, in your tests, you can use a single precision floating point value to represent distance since it is the primary measurement for this task. Additional information for the report message type regarding lava tunnel distance might include the combination of this distance and other static information (e.g., `frame="lava_tube_entrance", distance=3.1415`).

**Reply by gijoop** [reply]
Thank you for your quick response! These details will certainly help us with preparing for autonomous task
